app/models/training_period.rb

```diff
@@ -49,6 +50,18 @@ class TrainingPeriod < ApplicationRecord
     left_outer_joins(:ect_at_school_period, :mentor_at_school_period)
       .merge(ECTAtSchoolPeriod.for_school(school).or(MentorAtSchoolPeriod.for_school(school)))
   }
+
+  scope :latest_ect_training_period, ->(teacher:, lead_provider:) {
```

Comment:
Would this make more sense as something like `ect_training_periods_latest_first` so its clear it returns multiple?

Reply:
@peteryates we can't make these `has_one` as we need to filter by lead provider